internal/controllers/networkpolicy.go

```diff
@@ -176,7 +178,9 @@ func (r *Reconciler) reconcileStorageNetworkPolicy(ctx context.Context, cr *mode
 					},
 					Ports: []networkingv1.NetworkPolicyPort{
 						{
-							Port: &intstr.IntOrString{IntVal: constants.StoragePort},
+							Port: &intstr.IntOrString{
+								IntVal: getServicePortOrDefault(*serviceSpecs.StorageURL, constants.StoragePort),
```

Comment:
The network policies work using the container ports, right? If so, referencing the service port here may not be what we want.